test/test_traits.cpp

```diff
@@ -0,0 +1,28 @@
+#include "doctest/doctest.h"
+
+#include <sparrow/data_traits.hpp>
+
+/////////////////////////////////////////////////////////////////////////////////////////
+// Opt-in support for custom C++ representations of arrow data types.
+
+struct MyDataType{};
+
+template<>
+struct sparrow::arrow_traits<MyDataType>
+{
+    static constexpr data_type type_id = sparrow::data_type::INT32;
+    using value_type = MyDataType;
+    using default_layout = sparrow::fixed_size_layout<MyDataType>;
+};
+
+
+static_assert( sparrow::is_arrow_traits< sparrow::arrow_traits<MyDataType> > );
+static_assert( sparrow::any_arrow_type< MyDataType > );
+
+//////////////////////////////////////////////////////////////////////////////////////////
+// Base arrow types representations support tests and concept checking.
+namespace sparrow
+{
+    static_assert(mpl::all_of(all_base_types_t{}, predicate::is_arrow_base_type));
+    static_assert(mpl::all_of(all_base_types_t{}, predicate::has_arrow_traits));
```

Comment:
I guess this check is redundant, but we better keep it.

Reply:
These checks do not have the same meaning:
- the first one checks that every type listed is an arrow type, for the base types that means you might not need traits available at compile time (no need to include `data_traits.hpp` for base types to run this);
- the second check verifies that valid traits exists for each type of the list, but that does not guarantee that it's a valid arrow-supported type (although currently it is, we could add more conditions to `is_arrow_base_type` in the future).

Reply:
OK, so definitely let's keep it. :)